app/assets/javascripts/services/caseCSVSvc.js

```diff
@@ -351,35 +377,41 @@
          */
 
         function stringifySnapshot (aCase, snapshot, withHeader) {
-          var csvContent  = '';
-
+          let infoArray = [];
+          let matchingQuery = null;
+          let matchingQueryText = null;
+          const snapshotName = snapshot.name();
+          const snapshotTime = snapshot.time;
+          const caseNumber = aCase.caseNo;
+          let csvContent = '';
+          let queryIdAsInt = 0;
 
           if (withHeader) {
             csvContent += self.snapshotHeaderToCSV();
           }
 
-          angular.forEach(snapshot.docs, function (docs,queryId) {
-            angular.forEach(docs, function (doc,idx) {
-              var dataString;
-              var infoArray = [];
-
-              infoArray.push(stringifyField(snapshot.name()));
-              infoArray.push(stringifyField(snapshot.time));
-              infoArray.push(stringifyField(aCase.caseNo));
-
-              queryId = parseInt(queryId,10);
-              var query = snapshot.queries.filter(function(q) { return q.queryId === queryId; })[0];
-
-              infoArray.push(stringifyField(query.query_text));
-              infoArray.push(stringifyField(doc.id));
-              infoArray.push(stringifyField(idx+1));
-
-              dataString = infoArray.join(',');
-              csvContent += dataString + EOL;
+          angular.forEach(snapshot.docs, function (docs, queryId) {
+            queryIdAsInt = parseInt(queryId, 10);
```

Comment:
i think this should be where the "let" happens

Reply:
queryId is not known until this line, comes in from L393.  If you mean that you want to add a let at the beginning of L394 and take out L387, that would work.